stonesoup/tracker/pointprocess.py

```diff
@@ -0,0 +1,155 @@
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.gaussian_mixture is None:
+            self.gaussian_mixture = GaussianMixture()
+
+    @property
+    def tracks(self):
+        tracks = set()
+        for track in self.target_tracks.values():
+            tracks.add(track)
+        return tracks
+
+    def update_tracks(self):
+        """
+        Updates the tracks (:class:`Track`) associated with the filter.
+
+        Parameters
+        ==========
+        self : :class:`GaussianMixtureMultiTargetTracker`
+            Current GM Multi Target Tracker at time :math:`k`
+
+        Note
+        ======
+        Each track shares a unique tag with its associated component
+        """
+        for component in self.gaussian_mixture:
+            tag = component.tag
+            if tag != 0:
+                # Sanity check for birth component
+                if tag in self.target_tracks:
+                    # Track found, so update it
+                    track = self.target_tracks[tag]
+                    track.states.append(component)
+                else:
+                    # No Track found, so create a new one only if we are
+                    # reasonably confident its a target
+                    if component.weight > \
+                            self.extraction_threshold:
+                        self.target_tracks[tag] = Track([component], id=tag)
+
+    @BufferedGenerator.generator_method
+    def tracks_gen(self):
+        for time, detections in self.detector:
+            # Add birth component
+            self.birth_component.timestamp = time
+            self.gaussian_mixture.append(self.birth_component)
+            # Perform GM Prediction and generate hypotheses
+            hypotheses = self.hypothesiser.hypothesise(
+                        self.gaussian_mixture.components,
+                        detections,
+                        time
+                        )
+            # Perform GM Update
+            self.gaussian_mixture = self.updater.update(hypotheses)
+            # Reduce mixture - Pruning and Merging
+            self.gaussian_mixture.components = \
+                self.reducer.reduce(self.gaussian_mixture.components)
+            # Update the tracks
+            self.update_tracks()
+            self.end_tracks()
+            yield time, self.tracks
+
+    def end_tracks(self):
+        """
+        Ends the tracks (:class:`Track`) that do not have an associated
+        component within the filter.
+
+        Parameters
+        ==========
+        self : :class:`GaussianMixtureMultiTargetTracker`
+            Current GM Multi Target Tracker at time :math:`k`
+        """
+        component_tags = [component.tag for component in self.gaussian_mixture]
+        track_indices_to_delete = [tag for tag in self.target_tracks if tag not in component_tags]
+        # Delete the track
+        for key in track_indices_to_delete:
```

Comment:
Sets would be better here, as you can just take the difference, and dictionary key views act like a set:
```python
component_tags = {component.tag for component in self.gaussian_mixture}
# Delete the track
for key in self.target_tracks.keys() - component_tags:
```